hawc/apps/animalv2/views.py

```diff
@@ -0,0 +1,343 @@
+from django.db import transaction
+from django.db.models import Q
+from django.forms import modelformset_factory
+from django.http import HttpRequest
+from django.shortcuts import render
+
+from ..common.htmx import HtmxViewSet, Item, action, can_edit, can_view
+from ..common.views import (
+    BaseCreate,
+    BaseDelete,
+    BaseDetail,
+    BaseUpdate,
+    FormsetConfiguration,
+    create_object_log,
+)
+from ..mgmt.views import EnsureExtractionStartedMixin
+from ..study.models import Study
+from . import forms, models
+
+
+# Experiment Views
+class ExperimentCreate(EnsureExtractionStartedMixin, BaseCreate):
+    success_message = "Experiment created."
+    parent_model = Study
+    parent_template_name = "study"
+    model = models.Experiment
+    form_class = forms.ExperimentForm
+
+
+class ExperimentUpdate(BaseUpdate):
+    success_message = "Experiment updated."
+    parent_model = Study
+    parent_template_name = "study"
+    model = models.Experiment
+    form_class = forms.ExperimentForm
+    template_name = "animalv2/experiment_update.html"
+
+    # currently just using default behavior; once I am making sub-objects (animalgroups, treatments,
+    # etc.) then revisit ExperimentManager to do some smart prefetching. See
+    # epiv2/managers.py::DesignManager for an example of this in use.
+    # ExperimentDelete has a similarly commented-out overridden get_queryset...
+    """
+    def get_queryset(self):
+        return super().get_queryset().complete()
+    """
+
+
+class ExperimentDetail(BaseDetail):
+    model = models.Experiment
+
+
+class ExperimentDelete(BaseDelete):
+    success_message = "Experiment deleted."
+    model = models.Experiment
+
+    """
```

Comment:
remove commented out code or add back in with a TODO comment

Reply:
done